### DISCUSSION TITLE
Saber Presets and simple menu design.

### DISCUSSION THREAD

**Discussion by Mawntee** [ToniMacaroni/SaberFactory#139] [Ideas]
I really want to like SF as it adds many more options, and the ability to pick up the saber and see how everything looks in front of you is incredible, but it's really lacking a lot of basic features which is keeping me on the old and janky "Custom Sabers" mod. 

The first thing CS has going for it is the presets/profiles. 
![image](https://user-images.githubusercontent.com/51247415/184408465-d0cf5139-23a8-4a57-833d-f6952a18ce01.png)
I can pick a blade, pick a trail, adjust all the sizes and offsets of everything, then set it as a profile where I can quickly jump back to it later. 
An example of this is I have a "main" profile that's a little flashier for casual songs and when I'm streaming, a "tryhard" profile that has less intrusive trails and simple saber design, and then a "testplay" saber that has a different blade that shows rotation axis and keeps the same main trail that I use. It takes a single click to just jump from one to the next and I don't have to try and remember what specific trail offset and length I had set up for whatever setup I used in each condition. 

As for the menu I love the design and style of SF, it looks incredible don't get me wrong...
but it's weird that every mod uses the exact same standard menu design that everyone is familiar with, except for SF, and the menu doesn't really offer any extra ease of use even though it has a full re-design. It's a whole new environment that I need to learn how to navigate for no real added benefit other than aesthetics.

In CS you open it and everything is where you expect it to be
![image](https://user-images.githubusercontent.com/51247415/184409455-8e761aa7-6448-4c17-980d-f0d08db387ae.png)
- Settings/adjustments on the left
- Blade and trail options both visible on the same screen in the middle - up to 7 at a time (x2 for trail options also visible on same menu)
- Preview of blades on the right, as well as in the players hand to preview trail settings. 

In SF you have a brand new foreign menu (which looks sick AF, but is kinda confusing)
![image](https://user-images.githubusercontent.com/51247415/184410039-cab460ce-0fe9-4de8-8392-2de51956f6de.png)
- New icons with no (clearly visible) text descriptions on the left - The player just has to kinda hover, wait, and "figure it out"
- Significantly taller menu design that still somehow only holds the same 7 options
- That menu only shows single options of either blade, or trail, making it difficult try mix and match.
- Scroll arrows and reload/delete buttons are placed in a really weird spot imho and take up much more space than necessary. 
- When switching to trail options the "grab" button is removed, so the player has to jump back a menu just to preview the trail properly.

Honestly there's just so many unnecessary clicks and second guessing just to do something that has been considered "basic" for a long time already. 

Like why do I need to make a whole click and get away from the main selection menu, then click back again just to adjust these three simple settings?
![image](https://user-images.githubusercontent.com/51247415/184411329-be29d617-bafc-4270-ba7a-3b2f5c2e881c.png)
This also seems incredibly odd because "Saber Width" is tucked below the main menu options??
![image](https://user-images.githubusercontent.com/51247415/184411450-13a08d9f-8016-4594-878a-8f2b371e79d9.png)
Why is that the only transform option easily accessible, why is that not tied in with the others?

TL;DR: I love the features, everything looks rad, but it's incredibly over engineered to the point that it's somehow more difficult to use as an end user. imho there's way too much unused space in the scene that can simplify a lot of things.

**Comment by ToniMacaroni** [comment]
Hi Mawntee,
Thanks for the thorough critisism.

As a first quick note the concept of presets was in SF from the beginning but V2 never saw the UI implementation of it. I already integrated full preset support in V3 as it will go very well with the feature to have seperate presets for desktop/hmd view.

For the UI itself let me explain some of the design choices.

The rotation, z offset and saber length settings are hidden on another screen because people barely use them (including myself). Those are settings that most of the time can be left at default if the author of the saber exported it correctly. The saber width option on the other hand is a very personal setting and most of the time the author can't really ship the saber with a "perfect" width as it will vary from user to user.
That's why the setting is as quickly accessible as possible.

The reason a lot of icons are visible on the screen is, cause I think they are a great way to reduce clutter in the UI (especially in VR) and make them faster to notice. Granted they aren't the most descriptable icons (I changed them a bit in V3) but you only ever need to learn them once.

As for the "narrow but long" UI the reason is I really want the saber preview to be cleary visible in front of the player. And a wide (normal) UI would be obscured by the saber preview in most cases.

Now the mentioned decisions sound a lot like I was on a mission to make the UI as "simple and clean as possible". And that is exactly the case. The reason for that is, V1 was a lot like CS in that a lot of information was present at once on one big screen. Since a lot of people felt overwhelmed with this design, I tried to make a simpler alternative in V2. And since I haven't gotten any critisism with the new design I kept it.

All of this doesn't mean I'm happy with how the UI works now. Quite the contrary, that's why I change stuff around almost every release.
Also keep in mind you have probably been using CS for a long time now and maybe it's your first time using SF. They are different mods, they work in a different way and they came from different backgrounds.

But if you used SF for a bit and still aren't happy about certain things I will be happy to accept proposals for a new UI layout (could be even in the form of an MSPaint sketch).
I think it's save to say that a lot of SF how it is now came from user suggestions. I've been implementing most of the requested features no matter how absurd to difficult they were to implement.
If you have a feature or a change that would make a lot of sense in this mod, there's a good chance I will implement it.
